### DISCUSSION TITLE
LDAP Authentication

### DISCUSSION THREAD

**Discussion by Tithras** [jchristn/WatsonTcp#194] [General]
Good morning, does WatsonTcp have any LDAP support?

I am looking to try to authenticate that a user is allowed to send a message to all clients as well as have the client authenticate that message that it just received is authenticated and can be shown.

Currently I am doing the send via an xml which just says admin or user and if it the server sees admin then it can pass along the message. The client has nothing on it when it receives so in theory anyone could send a correctly formed message to the client on its listening port (i assume at least). I have looked at the encryption but run into how to transfer the key around so was looking to a stop gap rather than having to figure out that aspect of coding and key management!

Thanks.

Tithras

**Comment by jchristn** [comment]
Hi @Tithras sorry, authentication with an external identity service is out of scope for the library.  Cheers, Joel

**Reply by Tithras** [reply]
no probs thanks :)
